src/Language/PureScript/TypeChecker/Subsumption.hs

```diff
@@ -68,18 +68,18 @@ subsumes' val (TypeApp f1 r1) (TypeApp f2 r2) | f1 == tyRecord && f2 == tyRecord
   go ((p1, ty1) : ts1) ((p2, ty2) : ts2) r1' r2'
     | p1 == p2 = do _ <- subsumes Nothing ty1 ty2
                     go ts1 ts2 r1' r2'
+    | p1 /= p2 && r1' == REmpty && p2 `notElem` map fst ts1 = 
```

Comment:
We should be able to use sortedness here to avoid the `O(n^2)` complexity introduced by calling `notElem` at each element.